src/controllers/groupe.controller.ts

```diff
@@ -0,0 +1,83 @@
+import { NextFunction, Request, Response } from 'express';
+import { Group } from '@prisma/client';
+import { CreateGroupDto } from '@dtos/group.dto';
+import groupService from '@services/groups.service';
+
+class GroupController {
+  public groupService = new groupService();
+
+  public getGroups = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+    try {
+      const findAllGroupsData: Group[] = await this.groupService.findAllGroup();
+
+      res.status(200).json({ data: findAllGroupsData, message: 'findAll' });
+    } catch (error) {
+      next(error);
+    }
+  };
+
+  public getGroupById = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+    try {
+      const groupId = Number(req.params.id);
+      const findOneGroupData: Group = await this.groupService.findGroupById(groupId);
+
+      res.status(200).json({ data: findOneGroupData, message: 'findOne' });
+    } catch (error) {
+      next(error);
+    }
+  };
+
+  public createGroup = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+    try {
+      const groupData: CreateGroupDto = req.body;
+      const createGroupData: Group = await this.groupService.createGroup(groupData);
```

Comment:
what if the group already exists? you need to check if the group exists, if it does throw a `409 conflict`

```suggestion
   const group: Group = await this.groupService.find(...);
   if(group) { ... throw here }

  const createGroupData: Group = await this.groupService.createGroup(groupData);

```